### DISCUSSION TITLE
How to change the order of $FPATH?

### DISCUSSION THREAD

**Discussion by yibenhuang** [marlonrichert/zsh-snap#220] [Q&A] [answered]
Hi,

I have put the following lines into my .zshrc file,
```
znap source clarketm/zsh-completions
znap source conda-incubator/conda-zsh-completion
```
and this would result in the following `$FPATH` order
```
/Users/yiben/.znap/zsh-completions/src
/Users/yiben/.znap/conda-zsh-completion
```

Due to the outdated `_conda` in the `clarketm/zsh-completions` repo, I would like the one in the `conda-incubator/conda-zsh-completion` be used first. I tried to swap the order of sourcing them in the .zshrc file, but it didn't work.

Any help that how to change the order of `$FPATH`?

**Comment by marlonrichert** [comment, marked as answer]
Don't use `znap source` for completions or other kinds of functions. Do something like this instead:
```zsh
znap clone clarketm/zsh-completions conda-incubator/conda-zsh-completion
fpath=(
    ~[conda-incubator/conda-zsh-completion]
    ~[clarketm/zsh-completions]/src
    $fpath
)
```
This gives you full control of what goes in your `$fpath` and where.

**Reply by yibenhuang** [reply]
Thanks, good to know!
